README.md

```diff
@@ -1,2 +1,266 @@
-# lb4-log-extension-example
+# loopback4-example-log-extension
 An example repo showing how to write a complex log extension
+
+## Overview
+
+This repository shows you how to use [loopback4-extension-starter]() to write a complex logging extension that requires a [Component](), [Decorator](), and a [Mixin]().
+
+To use the extension, the user would load the component to get access to a `LogFn` that can be used in a sequence to log information. A Mixin allows the user to set the application wide logLevel. Only Controller methods configured at or above the logLevel will be logged. 
+
+Possible levels are: DEBUG < INFO < WARN < ERROR < OFF
+
+*Possible levels are represented as numbers but users can use `LOG_LEVEL.${level}` to specify the value instead of using numbers.*
+
+A decorator is provided to allow a user to provide metadata for Controller methods by using it to set the minimum logLevel the method should be logged at.
```

Comment:
A decorator enables you to provide metadata for Controller methods to set the minimum logLevel.